### PR TITLE
[mesh] [feature] Support for different active scalar and vector dataset groups

### PR DESCRIPTION
**Pull request by PeterPetrik**
vectors_contours_mesh

heavily inspired by crayfish implementation https://github.com/lutraconsulting/qgis-crayfish-plugin/blob/master/crayfish/gui/dataset_view.py

### CODE REVIEW

File: external/mdal/frmts/mdal_binary_dat.cpp [hunk 1]
```diff
@@ -129,6 +129,12 @@ void MDAL::LoaderBinaryDat::load( MDAL::Mesh *mesh, MDAL_Status *status )
   group->uri = mDatFile;
   group->isOnVertices = true;
 
+  // in TUFLOW results there could be also a special timestep (99999) with maximums
+  // we will put it into a separate dataset
+  std::shared_ptr<DatasetGroup> groupMax( new DatasetGroup() );
```
Comment (nyalldawson) [inline external/mdal/frmts/mdal_binary_dat.cpp]: this should be std::make_shared - but I'll fix in mdal

**Review by nyalldawson — COMMENTED** [review]
Code looks good to me!

**Comment by PeterPetrik** [comment]
@nyalldawson do you know why the PyQgsWFSProvider test fails? It looks completely unrelated to what I did.

File: external/mdal/frmts/mdal_binary_dat.cpp (hunk 1 above)
Comment (PeterPetrik) [reply external/mdal/frmts/mdal_binary_dat.cpp]: thanks! is it worth to release new new MDAL and update here? I am confident that I need to update MDAL before qgis 3.4 release later when I add support to more formats.

**Comment by nyalldawson** [comment]
It's a flakey test (on Travis at least). Just ignore it

**Pull request merged**